mne/channels/montage.py

```diff
@@ -483,11 +487,6 @@ def read_dig_montage(hsp=None, hpi=None, elp=None, point_names=None,
 
     .. versionadded:: 0.9.0
     """
-    if not isinstance(unit, string_types) or unit not in('m', 'mm', 'cm'):
-        raise ValueError('unit must be "m", "mm", or "cm"')
```

Comment:
I would keep the sanity check. If I understand correctly, with this solution, data will just not be scaled if someone passes something erroneus (like ``"CM"``).